pkg/kibana/kibana_provider.go

```diff
@@ -94,14 +95,29 @@ func (kb KibanaProvider) DeleteService(auth common.IKubernetesAuthInformation, i
 }
 
 // Converts a v1.Kibana instance to an service reprkbentation
-func (kb KibanaProvider) CrdInstanceToServiceInstance(crdInstance *v1.Kibana) *service.IService {
+func (kb KibanaProvider) CrdInstanceToServiceInstance(auth common.IKubernetesAuthInformation, crdInstance *v1.Kibana) *service.IService {
 	yamlData, err := yaml.Marshal(crdInstance)
 	if err != nil {
 		yamlData = []byte("Unknown")
 	}
 
+	api, err := kubernetes.GenerateK8sApiFromToken(kb.Host, kb.CaPath, auth.GetKubernetesAccessToken())
+
+	if err != nil {
+		return nil
```

Comment:
If the api could not get created this will result in an Error at another point. Method should return the error instead.

Reply:
When we look the func GenerateK8sApiFromToken in pkg/kubernetes/K8sApi.go, it would return error if the api could not be created

Reply:
Yes, but if err != nil { return nil } will suppress this error and result in an error at another place

Reply:
So, u mean, we would return an empty service pointer for the case that error occurs ?

Reply:
I mean that the function should return an error s.t. the functions that calls this function can also return the error correctly.

Reply:
So, return []byte("Some error message")